src/webgpu/shader/execution/robust_access.spec.ts

```diff
@@ -62,35 +62,43 @@ fn main() {
 
   t.debug(source);
   const module = t.device.createShaderModule({ code: source });
-  const pipeline = await t.device.createComputePipelineAsync({
-    layout,
-    compute: { module, entryPoint: 'main' },
-  });
-
-  const group = t.device.createBindGroup({
-    layout: pipeline.getBindGroupLayout(1),
-    entries: [
-      { binding: 0, resource: { buffer: constantsBuffer } },
-      { binding: 1, resource: { buffer: resultBuffer } },
-    ],
-  });
 
-  const testGroup = t.device.createBindGroup({
-    layout: pipeline.getBindGroupLayout(0),
-    entries: testBindings,
-  });
+  try {
+    const pipeline = await t.device.createComputePipelineAsync({
+      layout,
+      compute: { module, entryPoint: 'main' },
+    });
 
-  const encoder = t.device.createCommandEncoder();
-  const pass = encoder.beginComputePass();
-  pass.setPipeline(pipeline);
-  pass.setBindGroup(0, testGroup, dynamicOffsets);
-  pass.setBindGroup(1, group);
-  pass.dispatchWorkgroups(1);
-  pass.end();
+    const group = t.device.createBindGroup({
+      layout: pipeline.getBindGroupLayout(1),
+      entries: [
+        { binding: 0, resource: { buffer: constantsBuffer } },
+        { binding: 1, resource: { buffer: resultBuffer } },
+      ],
+    });
 
-  t.queue.submit([encoder.finish()]);
+    const testGroup = t.device.createBindGroup({
+      layout: pipeline.getBindGroupLayout(0),
+      entries: testBindings,
+    });
 
-  t.expectGPUBufferValuesEqual(resultBuffer, new Uint32Array([0]));
+    const encoder = t.device.createCommandEncoder();
+    const pass = encoder.beginComputePass();
+    pass.setPipeline(pipeline);
+    pass.setBindGroup(0, testGroup, dynamicOffsets);
+    pass.setBindGroup(1, group);
+    pass.dispatchWorkgroups(1);
+    pass.end();
+
+    t.queue.submit([encoder.finish()]);
+    t.expectGPUBufferValuesEqual(resultBuffer, new Uint32Array([0]));
+  } catch (err) {
+    if (err instanceof GPUPipelineError) {
+      t.fail(`Pipeline Creation Error, ${err.reason}: ${err.message}`);
+    } else {
+      throw err;
+    }
+  }
```

Comment:
Nothing fundamentally wrong with this change, but it _is_ supposed to be totally fine to throw an exception from a test function (which awaiting a promise that rejects will do). I also don't want to cargo-cult this every place we use create*PipelineAsync if it's not necessary.